pkg/genericapiserver/enablement_config.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package genericapiserver
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+type APIResourceEnablement interface {
+	ResourceEnabled(resource unversioned.GroupVersionResource) bool
+	VersionEnabled(version unversioned.GroupVersion) bool
+}
+
+// Specifies the overrides for various API group versions.
+// This can be used to enable/disable entire group versions or specific resources.
+type APIGroupVersionEnablementConfig struct {
```

Comment:
> This is not used outside this file. Can it be a private type?

I hadn't ruled out using downstream.  It's a convenient impl for any importers who just want to try to re-use the `genericapiserver` without much customization, which I'm assuming most implementors will choose to do.  I'd like to keep it public unless you think its leaking too many details.

The `genericapiserver.Config` only takes the "read" side of the interface.  The setters only exist on this impl.